File: server/baked/src/main/java/com/example/baked/model/AuthUser.java
package com.example.baked.model;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document
@NoArgsConstructor
@AllArgsConstructor
@Data
public class AuthUser {
  @Id private String id;
  private String username;
  private String password;
  private ArrayList<Role> roles;
  private FullName fullname;
  private String gender;
  private Date dob;
  private Address address;
  private List<String> emails;
  private List<String> phones;
  private String about;
  private List<Relative> relatives;
  private Student student;
  private Tutor tutor;
}

File: client/brooks/lib/model/AuthUser.ts
export interface AuthUser {
  id: string;
  username: string;
  password: null;
  roles: string[];
  userMetadata: UserMetadata;
}

export interface UserMetadata {
  fullname: Fullname;
  gender: string;
  dob: string;
  address: Address;
  emails: string[];
  phones: string[];
  about: string;
  relatives: Relative[];
  student: Student;
  tutor: Tutor;
}

export interface Address {
  province_city: string;
  ward_district: string;
  home_number: string;
}

export interface Fullname {
  first_name: string;
  last_name: string;
}

export interface Relative {
  id: string;
  relationship: string;
  fullname: Fullname;
  gender: string;
  dob: string;
  address: Address;
  emails: string[];
  phones: string[];
  about: string;
}

export interface Student {
  classes: string[];
  periods: string[];
}

export interface Tutor {
  job: string;
  graduatedSchool: string;
  major: string;
  qualification: string;
  graduatedYear: number;
  grades: string[];
  subjects: string[];
  minimumSalaryRequirement: number;
  teachingClasses: string[];
  periods: string[];
}
